packages/excubiae/contracts/test/MockSemaphore.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import {ISemaphore} from "@semaphore-protocol/contracts/interfaces/ISemaphore.sol";
+
+/// @title Mock Semaphore Contract
+/// @notice This contract is a mock implementation of the ISemaphore interface for testing purposes.
+/// @dev It simulates the behavior of a real Semaphore contract by simulating the storage and verification
+/// of a set of predefined mocked proofs.
+contract MockSemaphore is ISemaphore {
+    /// @dev Gets a group id and returns the relative group.
+    mapping(uint256 => bool) public mockedGroups;
+
+    /// @notice A mapping to store mocked proofs by their unique nullifiers.
+    mapping(uint256 => bool) private mockedProofs;
+
+    /// @dev Counter to assign an incremental id to the groups.
+    /// This counter is used to keep track of the number of groups created.
+    uint256 public groupCounter;
+
+    /// MOCKS ///
+    /// @notice Constructor to initialize the mock contract with predefined proofs.
+    /// @param _groupIds An array of identifiers of groups to be intended as the contract managed groups.
+    /// @param _nullifiers An array of nullifiers to be mocked as proofs.
+    /// @param _validities An array of booleans to mock the validity of proofs associated with the nullifiers.
+    constructor(uint256[] memory _groupIds, uint256[] memory _nullifiers, bool[] memory _validities) {
```

Comment:
I know it's a mock so no difference really, but if you don't modify the parameters you can use calldata instead of memory

Reply:
correct, let me fix that!